misc/CMakeLists.txt

```diff
@@ -1,3 +1,5 @@
+add_definitions(-D__ZEPHYR_SUPERVISOR__)
```

Comment:
This is wrong. This negates the purpose of adding the syscalls in these files. 
These files have functions that will be called directly by the user. Then inside this we do syscalls. By Adding this macro the check for usersapce will be not done. And this is a security issue. We shouldn't be doing this.